src/Servant/OpenApi/Internal.hs

```diff
@@ -467,3 +470,155 @@ instance (ToResponseHeader h, AllToResponseHeader hs) => AllToResponseHeader (h
 
 instance AllToResponseHeader hs => AllToResponseHeader (HList hs) where
   toAllResponseHeaders _ = toAllResponseHeaders (Proxy :: Proxy hs)
+
+type DeclareDefinition = Declare (Definitions Schema)
+
+instance
+  (AllToResponseHeader hs, IsSwaggerResponse r) =>
+  IsSwaggerResponse (WithHeaders hs a r)
+  where
+  responseSwagger =
+    fmap
+      (headers .~ fmap Inline (toAllResponseHeaders (Proxy @hs)))
+      (responseSwagger @r)
+
+class IsSwaggerResponseList as where
+  responseListSwagger :: DeclareDefinition (InsOrdHashMap HttpStatusCode Response)
+
+class IsSwaggerResponse a where
+  responseSwagger :: DeclareDefinition Response
+
+simpleResponseSwagger :: forall a cs desc. (ToSchema a, KnownSymbol desc, AllMime cs) => DeclareDefinition Response
```

Comment:
Is there really no existing instance or function for this?

Reply:
I took this code from the Wire codebase. I'm not sure there's anything that already exist that could replace it.

Reply:
Well, ok. As long as it does not break anything :)